sculture-android/app/src/main/java/tr/edu/boun/cmpe/sculture/activity/ProfilePageActivity.java

```diff
@@ -0,0 +1,161 @@
+import com.android.volley.VolleyError;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import tr.edu.boun.cmpe.sculture.R;
+import tr.edu.boun.cmpe.sculture.adapter.StoryListViewAdapter;
+import tr.edu.boun.cmpe.sculture.models.response.BaseStoryResponse;
+import tr.edu.boun.cmpe.sculture.models.response.SearchResponse;
+
+import static tr.edu.boun.cmpe.sculture.BaseApplication.baseApplication;
+import static tr.edu.boun.cmpe.sculture.Constants.API_USER_STORIES;
+import static tr.edu.boun.cmpe.sculture.Constants.FIELD_ID;
+import static tr.edu.boun.cmpe.sculture.Constants.FIELD_PAGE;
+import static tr.edu.boun.cmpe.sculture.Constants.FIELD_SIZE;
+import static tr.edu.boun.cmpe.sculture.Constants.PREF_USER_ID;
+import static tr.edu.boun.cmpe.sculture.Utils.addRequest;
+
+public class ProfilePageActivity extends AppCompatActivity {
+
+    TextView username;
+    TextView email;
+    RecyclerView story_list_recycler;
+    RelativeLayout loggedInLayout;
+    Button follow;
+
+    LinearLayoutManager mLayoutManager;
+    StoryListViewAdapter mStoryListViewAdapter;
+
+    private int PAGE = 1;
+    private boolean is_loading_more = false;
+    private boolean is_reach_end = false;
+
+    long currentUserID;
+    long visitedUserID;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_profile_page);
+
+
+        loggedInLayout = (RelativeLayout) findViewById(R.id.profile_layout_loggedIn);
+
+        username = (TextView) findViewById(R.id.profile_username);
+        email = (TextView) findViewById(R.id.profile_email);
+
+        follow = (Button) findViewById(R.id.follow);
+
+        story_list_recycler = (RecyclerView) findViewById(R.id.profile_story_list);
+        mLayoutManager = new LinearLayoutManager(this);
+        mStoryListViewAdapter = new StoryListViewAdapter(this);
+        story_list_recycler.setLayoutManager(mLayoutManager);
+        story_list_recycler.setAdapter(mStoryListViewAdapter);
+
+        setRecyclerListeners();
+
+        load_story();
+
+
+        final JSONObject requestBody = new JSONObject();
+        try {
+            currentUserID = requestBody.getLong(PREF_USER_ID);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            JSONObject requestBody2 = new JSONObject(username.getText().toString());
+            visitedUserID = requestBody2.getLong(username.getText().toString());
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        if(currentUserID == visitedUserID)
+            follow.setVisibility(View.GONE);
+
+        follow.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                JSONObject rb = new JSONObject();
+                try {
+                    requestBody.put(API_USER_STORIES, visitedUserID);
```

Comment:
This requestBody is not the JSONObject which defined the previous line. I think you have made an error.
Also /user/follow API is exist. You can integrate it now.